src/kube.rs

```diff
@@ -279,28 +304,101 @@ async fn get_all_pod_info() -> Result<Vec<PodInfo>, Box<dyn ::std::error::Error>
     let mut lp = ListParams::default();
     lp.label_selector = LogRecorderConfig::global().selector.clone();
 
-    for p in pods.list(&lp).await? {
-        for c in p.spec.containers {
-            let container = c.name;
-            let pod_name = p.metadata.name.to_string();
-            let file_name = LogRecorderConfig::global().outfile.to_string()
-                + &pod_name
-                + "-"
-                + &container
-                + ".txt";
-
-            let pod_info = PodInfo {
-                name: pod_name,
-                container: container,
-                file_name: file_name,
-            };
-            pod_info_vec.push(pod_info);
+    
+    let found_pods = pods.list(&lp).await?;
+
+    if LogRecorderConfig::global().container.len() > 0 || LogRecorderConfig::global().pod.len() > 0 {
+        let mut pod_filter_map = HashMap::new();
+        let mut container_filter_map = HashMap::new();
+
+        let filter_pods = LogRecorderConfig::global().pod.to_vec();
+        let filter_containers = LogRecorderConfig::global().container.to_vec();
+
+        for filter_p in filter_pods {
```

Comment:
anyway we can change these into
`let pod_filter_map: HashMap<PodInfo, book> = filter_pods.iter().map(|filter_p| (filter_p.to_owned, true)).collect(); `

Reply:
sounds great. You mean a HashMap of type HashMap<String, bool> like I used in the latest commit?

https://github.com/ericmcbride/wufei/pull/54/files#diff-2f4d83be75b2d94f525e8d7735b7a89d5138f89eb434e762f41aae846270b948R316-R317